### PR TITLE
Chunk by tokens

### PR DESCRIPTION
**Pull request by llogiq**
This time, we won't try to split lines into zero and all the other tokens. :stuck_out_tongue:

I also added some assertions, which appear to fail for my tests, unfortunately.

### CODE REVIEW

**Comment by llogiq** [comment]
Now this looks like it should be working. In my tests chunking the bloop repo, I get *9984* chunks with the trivial by-line chunking without overlap and *8266* chunks by tokens with 50% overlap. However, we may still have problems, I get a lot of warnings from onnxruntime.

I may want to refactor the whole thing a bit to pull the embedding to the front of the process so we can reuse it after chunking. Otherwise I would appreciate some help debugging this.

File: server/bleep/src/semantic.rs [hunk 1]
```diff
@@ -186,9 +189,17 @@ impl Semantic {
...
+            repo_name,
+            relative_path,
+            buffer,
+            &*self.tokenizer,
+            self.config.embedding_input_size,
+            15,
+            self.config.overlap.unwrap_or(chunk::OverlapStrategy::Partial(0.5)),
+        );
+        let repo_plus_file = repo_name.to_owned() + "\t" + relative_path + "\n";
+        info!(chunk_count = chunks.len(), "found chunks");
```
Comment (ggordonhall) [inline server/bleep/src/semantic.rs:202]:
```suggestion
        debug!(chunk_count = chunks.len(), "found chunks");
```

File: server/bleep/src/semantic.rs [hunk 2]
```diff
@@ -199,7 +210,7 @@ impl Semantic {
                     "new chunk",
```
Comment (ggordonhall) [inline server/bleep/src/semantic.rs:210]: Do we still need this `debug` statement?

File: server/bleep/src/semantic/chunk.rs [hunk 3]
```diff
@@ -108,3 +378,49 @@ pub fn trivial(src: &str, size: usize) -> Vec<Chunk<'_>> {
...
+                &file.path().to_string_lossy(),
+                &src,
+                &tokenizer,
+                max_tokens,
+                max_lines,
+                OverlapStrategy::Partial(0.5),
+            )
+            .len();
+            let linewise = trivial(&src, 15).len();
+            dbg!((path, tokenwise, linewise));
```
Comment (ggordonhall) [inline server/bleep/src/semantic/chunk.rs:420]: Can we remove these `dbg!` statements?

**Review by ggordonhall — APPROVED** [review]
Lgtm. Just a couple of nits

File: server/bleep/src/semantic.rs (hunk 2 above)
Comment (llogiq) [reply server/bleep/src/semantic.rs]: No. Removing it.

**Review by rsdy — APPROVED** [review]

**Pull request merged**